src/server/services/RepositoryService.ts

```diff
@@ -648,4 +667,312 @@ export class RepositoryService implements IRepositoryService {
     sanitizedPath = path.format(parsedPath);
     return { sanitizedPath, originalPath };
   }
+
+  async initAllianceDatabase(): Promise<boolean> {
+    // if the database is already initialized
+    if (this.allianceDatabasePool != undefined) {
+      return true;
+    }
+
+    try {
+      this.allianceDatabasePool = await AllianceDatabase.getPool();
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  private async addToAllianceDatabase(
+    recompiledContract: CheckedContract,
+    match: Match
+  ) {
+    if (
+      recompiledContract.runtimeBytecode === undefined ||
+      recompiledContract.creationBytecode === undefined ||
+      match.deployedRuntimeBytecode === undefined ||
+      match.deployedCreationBytecode === undefined
+    ) {
+      // Can only store contracts with both runtimeBytecode and creationBytecode
+      return;
+    }
+
+    if (match.creatorTxHash === undefined) {
+      // Can only store matches with creatorTxHash
+      return;
+    }
```

Comment:
Uh oh. I wasn't aware of these requirements. It means we just can't send contracts most of the times

Reply:
I managed to get every time all the requirements. I forced to execute both deployed and creation verification every time.